examples/simulation/plot_source_simulator.py

```diff
@@ -0,0 +1,77 @@
+"""
+==============================
+Generate simulated source data
+==============================
+"""
+# Author: Kostiantyn Maksymenko <kostiantyn.maksymenko@gmail.com>
+#         Samuel Deslauriers-Gauthier <sam.deslauriers@gmail.com>
+#
+# License: BSD (3-clause)
+
+import os.path as op
+
+import numpy as np
+
+import mne
+from mne.datasets import sample
+
+print(__doc__)
+
+# For this example, we will be using the information of the sample subject.
+# This will download the data if it not already on your machine. We also set
+# the subjects directory so we don't need to give it to functions.
+data_path = sample.data_path()
+mne.set_config('SUBJECTS_DIR', op.join(data_path, 'subjects'))
```

Comment:
avoid this as this is a global change to the session. Pass subjects_dir explicitly
as parameter to objects/fucntions that need it